staging/src/k8s.io/apimachinery/pkg/apis/meta/v1/types.go

```diff
@@ -579,6 +579,8 @@ type APIResource struct {
 	// verbs is a list of supported kube verbs (this includes get, list, watch, create,
 	// update, patch, delete, deletecollection, and proxy)
 	Verbs Verbs `json:"verbs" protobuf:"bytes,4,opt,name=verbs"`
+	// shortName is the suggested short name of the resource.
+	ShortName string `json:"shortName,omitempty" protobuf:"bytes,5,opt,name=shortName"`
```

Comment:
Last time you thought that backward compatibility only mattered for scripts (which shouldn't use short names) and users could just be broken.  I like a list though, so if you've come around, then list it is. :)

For previous reference.  https://github.com/kubernetes/kubernetes/pull/38755#discussion_r92386603